packages/@aws-cdk/aws-redshift/lib/private/database-query-provider/table.ts

```diff
@@ -48,28 +71,79 @@ async function updateTable(
   tableNamePrefix: string,
   tableNameSuffix: string,
   tableColumns: Column[],
-  clusterProps: ClusterProps,
-  oldResourceProperties: TableHandlerProps & ClusterProps,
+  tableAndClusterProps: TableAndClusterProps,
+  oldResourceProperties: TableAndClusterProps,
 ): Promise<string> {
+  const alterationStatements: string[] = [];
+
   const oldClusterProps = oldResourceProperties;
-  if (clusterProps.clusterName !== oldClusterProps.clusterName || clusterProps.databaseName !== oldClusterProps.databaseName) {
-    return createTable(tableNamePrefix, tableNameSuffix, tableColumns, clusterProps);
+  if (tableAndClusterProps.clusterName !== oldClusterProps.clusterName || tableAndClusterProps.databaseName !== oldClusterProps.databaseName) {
+    return createTable(tableNamePrefix, tableNameSuffix, tableColumns, tableAndClusterProps);
   }
 
   const oldTableNamePrefix = oldResourceProperties.tableName.prefix;
   if (tableNamePrefix !== oldTableNamePrefix) {
-    return createTable(tableNamePrefix, tableNameSuffix, tableColumns, clusterProps);
+    return createTable(tableNamePrefix, tableNameSuffix, tableColumns, tableAndClusterProps);
   }
 
   const oldTableColumns = oldResourceProperties.tableColumns;
   if (!oldTableColumns.every(oldColumn => tableColumns.some(column => column.name === oldColumn.name && column.dataType === oldColumn.dataType))) {
-    return createTable(tableNamePrefix, tableNameSuffix, tableColumns, clusterProps);
+    return createTable(tableNamePrefix, tableNameSuffix, tableColumns, tableAndClusterProps);
```

Comment:
Can you educate me? What happens in this case? We have a table already with a name (e.g., MyTable) and we add a column. Then we execute a `CREATE TABLE MyTable...` statement with a slightly different set of columns. Wouldn't that fail due to the table name already existing? I'm wondering the same about all of the other `return createTable` calls further down this method.

Reply:
Yes the create table call would fail here in case the table needs replacing. The customer would then have a choice of manually dropping the table and/or updating the columns to avoid replacement.